tools/uiadm.sh

```diff
@@ -101,56 +127,68 @@ vtar() {
 info() {
 	ui="$INSTALL_PREFIX/bin/ui"
 	if [[ -f "$ui" ]]; then
-		echo "Version: $("$ui" version)"
+		echo "Version: $("$ui" version) $("$ui" stamp)"
+		# shellcheck disable=SC2154
 		echo "URL: https://$Admin_IP:4443"
 	else
 		echo "Not currently installed".
 	fi
 }
 
 get_avail_versions() {
-	vcurl "${URL_PREFIX}/?limit=1000" | \
-	  json -ga name | \
-	  sed -e 's/^smartos\-ui\-//' -e 's/\.tar\.gz$//' | \
-	  sort -t '.' -n -k1,1 -k2,2 -k3,3
-}
-
-avail() {
+	branch="$1"
 	ui="$INSTALL_PREFIX/bin/ui"
 	if [[ -f "$ui" ]]; then
-		version="$("$ui" version)"
-		get_avail_versions | grep -v "$version"
+		stamp="$("$ui" stamp)"
```

Comment:
Dumb question: Does this logic only show versions newer than currently-installed, and exclude the currently-installed?  (See  OS-8431  e.g.)

Reply:
Yeah, in a round-about way.

The marker in manta will only show objects listed after the marker, then we grep -v the currently installed version so that only newer versions are displayed.

It's relying on manta ordering, which IIRC is in the object create order (i.e., the order that rows were added to the table).

Reply:
Fascinating.  That's good to know.